tests/upgrades/test_subscription.py

```diff
@@ -30,161 +30,165 @@
 from robottelo.upgrade_utility import host_location_update
 
 
-@pre_upgrade
-def test_pre_manifest_scenario_refresh(request):
-    """Before upgrade, upload & refresh the manifest.
-
-    :id: preupgrade-29b246aa-2c7f-49f4-870a-7a0075e184b1
-
-    :steps:
-        1. Before Satellite upgrade, upload and refresh manifest.
-
-    :expectedresults: Manifest should be uploaded and refreshed successfully.
+class TestPreManifestScenarioRefresh:
+    """
+    The test class contains pre-upgrade and post-upgrade scenarios to test
```

Comment:
There is no purpose if we check it anywhere before the upgrade, but here we are executing our test cases on the new spin template(a base version of the satellite and capsule) and to avoid any environment-related issue this check is mandatory, Otherwise, we will be having doubt about pre-environment if something goes wrong.

Reply:
Make sense.